niceman/utils.py

```diff
@@ -895,4 +896,21 @@ def unicode(s):
     else:
         return __builtin__.unicode(s)
 
+
+def generate_unique_name(pattern, nameset):
+    """Create a unique numbered name from a pattern and a set"""
```

Comment:
a little bit more doc detail would be appreciated (e.g. Parameters   to describe that pattern is a string which will get "string interpolation with %" being applied given a single integer